openwisp_controller/config/whois/tasks.py

```diff
@@ -158,3 +167,88 @@ def delete_whois_record(ip_address):
     queryset = WHOISInfo.objects.filter(ip_address=ip_address)
     if queryset.exists():
         queryset.delete()
+
+
+@shared_task
+def manage_fuzzy_locations(
+    device_pk,
+    ip_address,
+    latitude=None,
+    longitude=None,
+    address=None,
+    add_existing=False,
+):
+    """
+    Creates/updates fuzzy location for a device based on the latitude and longitude
+    or attaches an existing location if `add_existing` is True.
+    Existing location here means a location of another device whose last_ip matches
+    the given ip_address.
+
+    - If attaching existing location and current device has no location then the
+    existing location is attached to the current device as well.
+    - If attaching existing location and current device already has location, then
+    current device's location is deleted and set same as existing device's location.
+    - If not attaching existing then new location is created if no location exists for
+    current device, or existing one is updated if it is approximate(fuzzy).
```

Comment:
Maybe, we can re-write this comment to something more readable like
```suggestion
    - If current device already has a location previously, then it is deleted.
    - If existing location is not attached, then a new location is created if no location exists for 
    current device, or existing one is updated if it is approximate(fuzzy).
```